openmc/mesh.py

```diff
@@ -799,6 +799,49 @@ def write_data_to_vtk(self, filename, datasets, volume_normalization=True):
             volume_normalization=volume_normalization
         )
 
+    def plot_tally_data_slice(self, dataset, basis: str='xy', axes=None):
+        """Maps the dataset values to the mesh and exports an image.
+
+            dataset : np.ndarray
+                A 2-D array of values to plot, this is rotated to suit the
+                MatPlotLib imshow function.
+            basis : {'xy', 'xz', 'yz'}
+                The basis directions for the plot
+            axes : matplotlib.Axes
+                Axes to draw to
+
+        Returns
+        -------
+        matplotlib.image.AxesImage
+            Resulting image 
+        """
+
+        import matplotlib.pyplot as plt
+        from matplotlib.colors import LogNorm
+
+        if basis == "xy":
+            oriented_data = np.rot90(dataset, 1)
+            xlabel, ylabel = 'x [cm]', 'y [cm]'
+        elif basis == "yz":
+            oriented_data = dataset
+            xlabel, ylabel = 'y [cm]', 'z [cm]'
+        else:  # basis == "xz"
+            oriented_data = np.rot90(dataset, -1)
+            xlabel, ylabel = 'x [cm]', 'z [cm]'
+
+        if axes is None:
+            fig, axes = plt.subplots()
+            axes.set_xlabel(xlabel)
+            axes.set_ylabel(ylabel)
+
+        return axes.imshow(
```

Comment:
Can we return the `axes` object here as well so users have access to it in the case that it was created by this method?

Reply:
Yes good idea